### PR TITLE
8269984: [macos] JTabbedPane title looks like  disabled

### PR DESCRIPTION
**Pull request by azuev-java**
Fixed the version string parsing to work correctly on macOS 11.x and beyond

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269984](https://bugs.openjdk.java.net/browse/JDK-8269984): [macos] JTabbedPane title looks like  disabled

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - Author)
 * [Prasanta Sadhukhan](https://openjdk.java.net/census#psadhukhan) (@prsadhuk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/269/head:pull/269` \
`$ git checkout pull/269`

Update a local copy of the PR: \
`$ git checkout pull/269` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/269/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 269`

View PR using the GUI difftool: \
`$ git pr show -t 269`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/269.diff">https://git.openjdk.java.net/jdk17/pull/269.diff</a>

</details>

### CODE REVIEW

**Review by kevinrushforth — APPROVED** [review]
Looks good. I also tested it on macOS 11.5 and verified that the test case fails (is rendered incorrectly) without your fix and passes with your fix.

**Comment by kevinrushforth** [comment]
@prsadhuk Can you review this?

File: src/java.desktop/macosx/classes/apple/laf/JRSUIUtils.java [hunk 1]
```diff
@@ -39,40 +39,55 @@
     static boolean isBigSurOrAbove = isMacOSXBigSurOrAbove();
```
Comment (prsadhuk) [inline src/java.desktop/macosx/classes/apple/laf/JRSUIUtils.java:39]: This is not required..Need to be removed.

**Review by prsadhuk — COMMENTED** [review]
Since existing currentMacOSXVersionMatchesGivenVersionRange() is changed, did you test in osx 10.15 or below to see current functionality is ok?
ALso, please add noreg-demo to JBS since I guess SwingSet2 is used to test.

**Comment by azuev-java** [comment]
> Since existing currentMacOSXVersionMatchesGivenVersionRange() is changed, did you test in osx 10.15 or below to see current functionality is ok?

I have tested the public methods of this class with simulated version strings for previous macOS releases - they return correct assumptions about them. I haven't tested the code that acquires the version string but i haven't changed that.

> ALso, please add noreg-demo to JBS since I guess SwingSet2 is used to test.

Done.

File: src/java.desktop/macosx/classes/apple/laf/JRSUIUtils.java (hunk 1 above)
Comment (azuev-java) [reply src/java.desktop/macosx/classes/apple/laf/JRSUIUtils.java]: > This is not required..Need to be removed.

Done.

**Review by prsadhuk — APPROVED** [review]

**Review by kevinrushforth — APPROVED** [review]
Updated fix looks good.

> > ALso, please add noreg-demo to JBS since I guess SwingSet2 is used to test.
> 
> Done.

That's not what `noreg-demo` means (that label means that the change itself is limited to a demo).

**Pull request closed without merging**